quokka_editor_front/src/components/molecules/deleteDocument/DeleteDocument.tsx

```diff
@@ -1,20 +1,35 @@
 import axios from "axios";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { API_URL } from "../../../consts";
-import { useAppDispatch } from "../../../Redux/hooks";
+import { useAppDispatch, useAppSelector } from "../../../Redux/hooks";
 import { deleteDocument } from "../../../Redux/documentsSlice";
 import Modal from "../../misc/Modal";
+import DocumentButton from "../../atoms/documentButton/DocumentButton";
 
-interface DeleteDocumentProps {
+interface DeleteDocumentType {
   id: string;
   title: string;
 }
 
-const DeleteDocument: React.FC<DeleteDocumentProps> = ({ id, title }) => {
+const DeleteDocument: React.FC = () => {
+  const documentsState = useAppSelector((state) => state.documents.documents);
   const dispatch = useAppDispatch();
   const [deleteModal, setDeleteModal] = useState(false);
+  const [documentToDelete, setDocumentToDelete] = useState<
+    DeleteDocumentType | undefined
+  >(undefined);
```

Comment:
Why undefined?

Reply:
Because Array.find() function used in line 23 returns undefined when nothing was found with given predicate.
Then in lines 50 and 52 I'm setting styles to delete button if documentToDelete is set or is undefined